test/attr/attr_extern.swift

```diff
@@ -37,5 +37,71 @@ func f11Scope() {
     func f11Inner()
 }
 
-@_extern(invalid, module: "m12", name: "f12") // expected-error {{expected '_extern' option such as 'wasm'}}
+@_extern(invalid, module: "m12", name: "f12") // expected-error {{expected '_extern' option such as 'c'}}
 func f12InvalidLang() // expected-error {{expected '{' in body of function declaration}}
+
+@_extern(c, "valid")
+func externCValid()
+
+@_extern(c, "") // expected-error {{expected non-empty C name in @_extern attribute}}
+func emptyCName()
+
+@_extern(c)
+func omitCName()
+
+@_extern(c, ) // expected-error {{expected string literal in '_extern' attribute}}
+func editingCName() // expected-error {{expected '{' in body of function declaration}}
+
+struct StructScopeC {
+    @_extern(c, "member_decl") // expected-error {{@_extern attribute can only be applied to global functions}}
+    func memberDecl()
+
+    @_extern(c, "static_member_decl")
+    static func staticMemberDecl()
+}
+
+func funcScopeC() {
+    @_extern(c, "func_scope_inner")
+    func inner()
+}
+
+@_extern(c, "c_value") // expected-error {{@_extern may only be used on 'func' declarations}}
+var nonFunc: Int = 0
+
+@_extern(c, "with_body")
+func withInvalidBody() {} // expected-error {{unexpected body of function declaration}}
+
+@_extern(c, "duplicate_attr_c_1")
+@_extern(c, "duplicate_attr_c_2") // expected-error {{duplicate attribute}}
+func duplicateAttrsC()
+
+@_extern(wasm, module: "dup", name: "duplicate_attr_wasm_1")
+@_extern(wasm, module: "dup", name: "duplicate_attr_wasm_2") // expected-error {{duplicate attribute}}
+func duplicateAttrsWasm()
+
+@_extern(c, "mixed_attr_c")
+@_extern(wasm, module: "mixed", name: "mixed_attr_wasm")
+func mixedAttrs_C_Wasm()
+
+class NonC {}
+@_extern(c)
+func nonCReturnTypes() -> NonC // expected-error {{'NonC' cannot be represented in C}}
+@_extern(c)
+func nonCParamTypes(_: Int, _: NonC) // expected-error {{'NonC' cannot be represented in C}}
+
+@_extern(c)
+func asyncFuncC() async // expected-error {{async functions cannot be represented in C}}
+
+@_extern(c)
+func throwsFuncC() throws // expected-error {{raising errors from C functions is not supported}}
+
+@_extern(c)
+func genericFuncC<T>(_: T) // expected-error {{'T' cannot be represented in C}}
+
```

Comment:
What happens with these validation cases where you have e.g.
```
@_extern(c)
@_extern(wasm, module: "foo", name: "bar")
func nonCReturnTypesMixed() -> NonC
```
or even just
```
@_extern(wasm, module: "foo", name: "bar")
func nonCReturnTypesWasm() -> NonC
```
Are there return types Wasm doesn't support? Do we get an error message about the type not being representable in C in that case?

Reply:
Added test cases